src/Hl7.Fhir.STU3/Model/ValueSetExpansionExtensions.cs

```diff
@@ -34,20 +33,68 @@ public static ValueSet.ContainsComponent FindCode(this ValueSet.ContainsComponen
             else
                 return null;
         }
+
+        internal static CodeSystem.ConceptDefinitionComponent? FindCode(this IEnumerable<CodeSystem.ConceptDefinitionComponent> concepts, string code)
+        {
+            foreach (var concept in concepts)
+            {
+                var predicate = () => concept.Code == code;
+                var result = concept.findCodeByPredicate(predicate);
+                if (result != null) return result;
+            }
+            return null;
+        }
+
+        private static CodeSystem.ConceptDefinitionComponent? findCodeByPredicate(this IEnumerable<CodeSystem.ConceptDefinitionComponent> concepts, Func<bool> predicate)
+        {
+            foreach (var concept in concepts)
+            {
+                var result = concept.findCodeByPredicate(predicate);
+                if (result != null) return result;
+            }
+            return null;
+        }
+
+        private static CodeSystem.ConceptDefinitionComponent? findCodeByPredicate(this CodeSystem.ConceptDefinitionComponent concept, Func<bool> predicate)
+        {
+            // Direct hit
+            if (predicate.Invoke())
+                return concept;
+
+            // Not in this node, but this node may have child nodes to check
+            if (concept.Concept != null && concept.Concept.Any())
+                return concept.Concept.findCodeByPredicate(predicate);
+            else
+                return null;
+        }
+
+
+        internal static List<CodeSystem.ConceptDefinitionComponent> FilterCodesByProperty(this IEnumerable<CodeSystem.ConceptDefinitionComponent> concepts, string property, DataType value)
+        {
+            Func<CodeSystem.ConceptDefinitionComponent, bool> predicate = concept => concept.Property.Any(p => p.Code == property && p.Value == value);
```

Comment:
This equals might not work - it does reference equality, so it will not work on two, say, FhirBooleans with the same bool value.